### ISSUE TITLE
2.24.x false positive for `no-named-as-default`

### ISSUE THREAD

**Issue by k-funk** [import-js/eslint-plugin-import#2222]
Only started happening after upgrading eslint@7.29.0 to 7.32.0 and eslint-plugin-import@2.23.4 to 2.24.x

My code:
```
import Calendar from 'react-calendar'
// warning  Using exported name 'Calendar' as identifier for default export  import/no-named-as-default
```

`node_modules/@types/react-calendar/index.d.ts`
```
export default function Calendar(props: CalendarProps): JSX.Element;
```

`node_modules/react-calendar/src/index.js`
```
export default Calendar;
```

```
├── @typescript-eslint/eslint-plugin@4.31.0
├── @typescript-eslint/parser@4.31.0
├── eslint-config-recommended@4.1.0
├── eslint-plugin-import@2.24.2
├── eslint-plugin-jest@24.4.0
├── eslint-plugin-react@7.25.1
├── eslint@7.32.0
├── typescript@4.4.2
```

**Comment by k-funk** [comment]
Actually, I tried pinning all my old versions, purging node_modules and removing package-lock.json, and I got the same false-positive. Some sub-dependency must be causing it.

Old, pinned dependencies:
```
├── @typescript-eslint/eslint-plugin@4.28.0
├── @typescript-eslint/parser@4.28.0
├── eslint-config-recommended@4.1.0
├── eslint-plugin-import@2.23.4
├── eslint-plugin-jest@24.4.0
├── eslint-plugin-react@7.25.1
```
